spec/ddtrace/contrib/sinatra/tracer_spec.rb

```diff
@@ -638,7 +631,7 @@
 
       it { is_expected.to be_ok }
 
-      it 'logs a warning' do
+      xit '[TODO:BROKEN:might not be possible anymore] logs a warning' do
```

Comment:
I an mot sure about that one, @marcotc hypothesised it might not be possible now.

Reply:
Hmm, I do see the log happening on rspec's STDOUT (even though the test case is `xit`'d):

```
Randomized with seed 3496
......**...**..................................................................................................**................................................**..........*W, [2022-01-24T11:23:36.521855 #15]  WARN -- ddtrace: [ddtrace] Sinatra integration is misconfigured, reported traces will be missing request metadata such as path and HTTP status code. Did you forget to add `register Datadog::Contrib::Sinatra::Tracer` to your `Sinatra::Base` subclass? See <https://docs.datadoghq.com/tracing/setup_overview/setup/ruby/#sinatra> for more details.
........................................**......................
```

Reply:
> I an mot sure about that one, @marcotc hypothesised it might not be possible now.

Do you mean that it is not possible to have a `modular app does not register the Datadog::Contrib::Sinatra::Tracer middleware` anymore?

Reply:
Found the issue, it was related to `Datadog.logger` instance being recreated between the time we setup up the test assertion vs execution time.